compiler/plugins/target/AMD-AIE/aie/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set(IREE_PACKAGE_ROOT_DIR "${CMAKE_CURRENT_LIST_DIR}")
+set(IREE_PACKAGE_ROOT_PREFIX "iree::target::amd-aie::aie")
+
+list(APPEND IREE_COMPILER_TABLEGEN_INCLUDE_DIRS
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include"
+    )
+
+
+iree_cc_library(
+  NAME
+    defs
+  INCLUDES
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include"
+)
+
+iree_cc_library(
+    NAME
+        AIEDialectIR
+    SRCS
+      ${IREE_MLIR_AIE_SOURCE_DIR}/lib/Dialect/AIE/IR/AIEDialect.cpp
+      ${IREE_MLIR_AIE_SOURCE_DIR}/lib/Dialect/AIE/IR/AIETargetModel.cpp
+    DEPS
+      ::defs
+      ::AIEDialectGen
+      ::AIEInterfaceGen
+      ::AIEOpsGen
+      ::AIETypesGen
+      ::AIEEnumsGen
+      ::AIEAttrsGen
+      MLIRIR
+)
+
+iree_tablegen_library(
+  NAME
+    AIEDialectGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIE.td"
+  OUTS
+    -gen-dialect-decls --dialect=AIE Dialect/AIE/IR/AIEDialect.h.inc
+    -gen-dialect-defs --dialect=AIE Dialect/AIE/IR/AIEDialect.cpp.inc
+
+)
+
+iree_tablegen_library(
+  NAME
+    AIEInterfaceGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIEInterfaces.td"
+  OUTS
+    -gen-op-interface-decls Dialect/AIE/IR/AIEInterfaces.h.inc
+    -gen-op-interface-defs Dialect/AIE/IR/AIEInterfaces.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIEOpsGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIEOps.td"
+  OUTS
+    -gen-op-decls Dialect/AIE/IR/AIEOps.h.inc
+    -gen-op-defs Dialect/AIE/IR/AIEOps.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIETypesGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIETypes.td"
+  OUTS
+    -gen-typedef-decls Dialect/AIE/IR/AIETypesGen.h.inc
+    -gen-typedef-defs Dialect/AIE/IR/AIETypesGen.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIEEnumsGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIEAttrs.td"
+  OUTS
+    -gen-enum-decls Dialect/AIE/IR/AIEEnums.h.inc
+    -gen-enum-defs Dialect/AIE/IR/AIEEnums.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIEAttrsGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIEAttrs.td"
+  OUTS
+    -gen-attrdef-decls Dialect/AIE/IR/AIEAttrs.h.inc -attrdefs-dialect=AIE
+    -gen-attrdef-defs Dialect/AIE/IR/AIEAttrs.cpp.inc -attrdefs-dialect=AIE
+)
+
+###############################################################################
+# AIEX Dialect
+###############################################################################
+
+iree_cc_library(
+    NAME
+        AIEXDialectIR
+    SRCS
+      ${IREE_MLIR_AIE_SOURCE_DIR}/lib/Dialect/AIEX/IR/AIEXDialect.cpp
+    DEPS
+    ::defs
+    ::AIEXDialectGen
+    ::AIEXOpsGen
```

Comment:
Just curious: is this sorcery or just that CMake allows `::` in target identifiers?

Reply:
I believe cmake allows it in identifiers and with some cmake magic, iree uses this to explicitly reference things elsewhere in the build hierarchy

Reply:
They are technically "ALIAS" targets and a small bit of string munging for the :: prefix case. CMake being CMake, they are a thing all their own vs unified with everything else. One power they have is that CMake will hard error if you refer to one that does not exist (i.e. vs trying to link to a library on the system with the name).